web_pwa_oca/static/src/js/worker/base/tools.js

```diff
@@ -0,0 +1,17 @@
+/* Copyright 2020 Tecnativa - Alexandre D. Díaz
+ * License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). */
+ "use strict";
+
+
+/**
+ *  Execute various promises in sequential order and return a promise with all results
+ *  Code from: https://decembersoft.com/posts/promises-in-serial-with-array-reduce/
+ *
+ * @param {Array[Promise]} tasks
+ * @returns {Promise[Array]}
+ */
+function execSequentialPromises(tasks) {
+    return tasks.reduce((promiseChain, currentTask) => {
+        return promiseChain.then(chainResults => currentTask.then(currentResult => [...chainResults, currentResult]));
```

Comment:
I observe you're using JS syntax that isn't supposed to exist on v12. Is that on purpose?

Reply:
Yes.
- Technically, workers are not part of Odoo.
- If you can run services workers, you can use ES6+ https://caniuse.com/serviceworkers
- Current infrastructure can't testing service workers :/
- To work with service workers you need use a lot of asynchronous calls... so use async/await makes the life more easier and the code more maintainable and readable = Use ES6+